src/fu-tool.c

```diff
@@ -548,10 +549,40 @@ fu_util_install_task_sort_cb (gconstpointer a, gconstpointer b)
 	return fu_install_task_compare (task1, task2);
 }
 
+static gboolean
+fu_util_download_out_of_process (const gchar *uri, const gchar *fn, GError **error)
+{
+	const gchar *argv[] = { "curl", uri, "--output", fn, NULL };
+	return fu_common_spawn_sync (argv, NULL, NULL, NULL, error);
+}
+
+static gchar *
+fu_util_download_if_required (FuUtilPrivate *priv, const gchar *perhapsfn, GError **error)
+{
+	g_autofree gchar *basename = NULL;
+	g_autofree gchar *filename = NULL;
+	g_autoptr(SoupURI) uri = NULL;
+
+	/* a local file */
+	uri = soup_uri_new (perhapsfn);
+	if (uri == NULL)
+		return g_strdup (perhapsfn);
+
+	/* download the firmware to a cachedir */
+	basename = g_path_get_basename (perhapsfn);
+	filename = g_build_filename (g_get_user_cache_dir (), "fwupdmgr", basename, NULL);
```

Comment:
s/fwupdmgr/fwupdtool/

Reply:
I did wonder about this. I was intending to use the same cache location between programs -- maybe both should be switched to `fwupd` rather than the command?

Reply:
Yeah I think that's a better idea

Reply:
Although... I think you want to move the cache that fwupdmgr uses for regular metadata and downloads over too then.  it currently uses `~/.cache/fwupdmgr/` too.

Which means maybe need to build a transitional symlink or rename if you move.